cuda_core/cuda/core/experimental/_program.py

```diff
@@ -6,6 +6,157 @@
+    builtin_move_forward: Optional[bool] = None  # /**< --builtin-move-forward={true|false} (-builtin-move-forward) Provide builtin definitions of std::move and std::forward. Default: true */
+    builtin_initializer_list: Optional[bool] = None  # /**< --builtin-initializer-list={true|false} (-builtin-initializer-list) Provide builtin definitions of std::initializer_list class and member functions. Default: true */
+    disable_warnings: Optional[bool] = None  # /**< --disable-warnings (-w) Inhibit all warning messages. */
+    restrict: Optional[bool] = None  # /**< --restrict (-restrict) Programmer assertion that all kernel pointer parameters are restrict pointers. */
+    device_as_default_execution_space: Optional[bool] = None  # /**< --device-as-default-execution-space (-default-device) Treat entities with no execution space annotation as __device__ entities. */
+    device_int128: Optional[bool] = None  # /**< --device-int128 (-device-int128) Allow the __int128 type in device code. */
+    optimization_info: Optional[str] = None  # /**< --optimization-info=<kind> (-opt-info) Provide optimization reports for the specified kind of optimization. */
+    display_error_number: Optional[bool] = None  # /**< --display-error-number (-err-no) Display diagnostic number for warning messages. Default: true */
+    no_display_error_number: Optional[bool] = None  # /**< --no-display-error-number (-no-err-no) Disable the display of a diagnostic number for warning messages. */
+    diag_error: Optional[str] = None  # /**< --diag-error=<error-number>,… (-diag-error) Emit error for specified diagnostic message number(s). */
+    diag_suppress: Optional[str] = None  # /**< --diag-suppress=<error-number>,… (-diag-suppress) Suppress specified diagnostic message number(s). */
+    diag_warn: Optional[str] = None  # /**< --diag-warn=<error-number>,… (-diag-warn) Emit warning for specified diagnostic message number(s). */
+    brief_diagnostics: Optional[bool] = None  # /**< --brief-diagnostics={true|false} (-brief-diag) Disable or enable showing source line and column info in a diagnostic. Default: false */
+    time: Optional[str] = None  # /**< --time=<file-name> (-time) Generate a CSV table with the time taken by each compilation phase. */
+    split_compile: Optional[int] = None  # /**< --split-compile= <number of threads> (-split-compile) Perform compiler optimizations in parallel. */
+    fdevice_syntax_only: Optional[bool] = None  # /**< --fdevice-syntax-only (-fdevice-syntax-only) Ends device compilation after front-end syntax checking. */
+    minimal: Optional[bool] = None  # /**< --minimal (-minimal) Omit certain language features to reduce compile time for small programs. */
+    device_stack_protector: Optional[bool] = None  # /**< --device-stack-protector (-device-stack-protector) Enable stack canaries in device code. */
+
+    def __post_init__(self):
+        # Format options into a list of strings
```

Comment:
I don't think this comment is correct.  Because of all the `.encode()` calls, won't `self.formatted_options` be a list of *bytes* objects?  Is that really intended?

Reply:
We could consider keeping strings and deferring the `.encode()` conversion to bytes in `Program` ctor. This way `ProgramOptions` can be easily inspected/debugged by human. WDYT?

Reply:
You're right about that being a bad comment. I will fix it. WRT your suggestion Leo, I used bytes so that we didn't need to duplicate the list of options, but perhaps that efficiency vs usability tradeoff is not worth it